drivers/sensor/pmw3360/pmw3360.c

```diff
@@ -766,158 +705,122 @@ static int pmw3360_async_init_configure(struct pmw3360_data *dev_data)
+						 init_work);
+	const struct device *dev = data->dev;
 
 	ARG_UNUSED(work);
 
-	LOG_DBG("PMW3360 async init step %d", dev_data->async_init_step);
+	LOG_DBG("PMW3360 async init step %d", data->async_init_step);
 
-	dev_data->err = async_init_fn[dev_data->async_init_step](dev_data);
-	if (dev_data->err) {
+	data->err = async_init_fn[data->async_init_step](dev);
+	if (data->err) {
 		LOG_ERR("PMW3360 initialization failed");
 	} else {
-		dev_data->async_init_step++;
+		data->async_init_step++;
 
-		if (dev_data->async_init_step == ASYNC_INIT_STEP_COUNT) {
-			dev_data->ready = true;
+		if (data->async_init_step == ASYNC_INIT_STEP_COUNT) {
+			data->ready = true;
 			LOG_INF("PMW3360 initialized");
 		} else {
-			k_work_schedule(&dev_data->init_work,
+			k_work_schedule(&data->init_work,
 					K_MSEC(async_init_delay[
-						dev_data->async_init_step]));
+						data->async_init_step]));
 		}
 	}
 }
 
-static int pmw3360_init_cs(struct pmw3360_data *dev_data)
-{
-	int err;
-
-	dev_data->cs_gpio_dev =
-		device_get_binding(PMW3360_CS_GPIO_DEV_NAME);
-	if (!dev_data->cs_gpio_dev) {
-		LOG_ERR("Cannot get CS GPIO device");
-		return -ENXIO;
-	}
-
-	err = gpio_pin_configure(dev_data->cs_gpio_dev, PMW3360_CS_GPIO_PIN,
-				 GPIO_OUTPUT);
-	if (!err) {
-		err = spi_cs_ctrl(dev_data, false);
-	} else {
-		LOG_ERR("Cannot configure CS PIN");
-	}
-
-	return err;
-}
-
-static int pmw3360_init_irq(struct pmw3360_data *dev_data)
+static int pmw3360_init_irq(const struct device *dev)
 {
 	int err;
+	struct pmw3360_data *data = dev->data;
+	const struct pmw3360_config *config = dev->config;
 
-	dev_data->irq_gpio_dev =
-		device_get_binding(PMW3360_IRQ_GPIO_DEV_NAME);
-	if (!dev_data->irq_gpio_dev) {
-		LOG_ERR("Cannot get IRQ GPIO device");
-		return -ENXIO;
+	if (!device_is_ready(config->irq_gpio.port)) {
+		LOG_ERR("IRQ GPIO device not ready");
+		return -ENODEV;
 	}
 
-	err = gpio_pin_configure(dev_data->irq_gpio_dev,
-				 PMW3360_IRQ_GPIO_PIN,
-				 GPIO_INPUT | GPIO_PULL_UP);
+	err = gpio_pin_configure_dt(&config->irq_gpio,
+				    GPIO_INPUT | GPIO_PULL_UP);
 	if (err) {
 		LOG_ERR("Cannot configure IRQ GPIO");
 		return err;
 	}
 
-	gpio_init_callback(&dev_data->irq_gpio_cb, irq_handler,
-			   BIT(PMW3360_IRQ_GPIO_PIN));
+	gpio_init_callback(&data->irq_gpio_cb, irq_handler,
+			   BIT(config->irq_gpio.pin));
 
-	err = gpio_add_callback(dev_data->irq_gpio_dev, &dev_data->irq_gpio_cb);
+	err = gpio_add_callback(config->irq_gpio.port, &data->irq_gpio_cb);
 	if (err) {
 		LOG_ERR("Cannot add IRQ GPIO callback");
 	}
 
 	return err;
 }
 
-static int pmw3360_init_spi(struct pmw3360_data *dev_data)
-{
-	dev_data->spi_dev = device_get_binding(PMW3360_SPI_DEV_NAME);
-	if (!dev_data->spi_dev) {
-		LOG_ERR("Cannot get SPI device");
-		return -ENXIO;
-	}
-
-	return 0;
-}
-
 static int pmw3360_init(const struct device *dev)
 {
-	struct pmw3360_data *dev_data = &pmw3360_data;
+	struct pmw3360_data *data = dev->data;
```

Comment:
We must initialize `data->dev` (`data->dev = dev`).

Reply:
Otherwise we will encounter BUS FAULT

Reply:
done